From 48c1bf02b31f04de060d96b7a74ccf655cec4feb Mon Sep 17 00:00:00 2001
From: Allen Ye <allen.ye@mediatek.com>
Date: Fri, 19 Dec 2025 14:49:09 +0800
Subject: [PATCH 253/254] mtk: hostapd: fix 2/5 GHz link start failed while
 enable unsol_bcast_probe_resp

Fix 2.4/5 GHz link start failed while enable unsol_bcast_probe_resp.
Due to hostapd doesn't gen 2/5 GHz UPR template, the
ubpr->unsol_bcast_probe_resp_tmpl sould be NULL.
In the case, we should not bring NL80211_ATTR_UNSOL_BCAST_PROBE_RESP
to mac80211 with no NL80211_UNSOL_BCAST_PROBE_RESP_ATTR_TMPL attr.

Signed-off-by: Allen Ye <allen.ye@mediatek.com>
---
 src/drivers/driver_nl80211.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/drivers/driver_nl80211.c b/src/drivers/driver_nl80211.c
index 252378a8e..f26941344 100644
--- a/src/drivers/driver_nl80211.c
+++ b/src/drivers/driver_nl80211.c
@@ -5785,7 +5785,8 @@ static int wpa_driver_nl80211_set_ap(void *priv,
 			goto fail;
 	}
 
-	if (params->ubpr.unsol_bcast_probe_resp_interval &&
+	if (params->freq && is_6ghz_freq(params->freq->freq) &&
+	    params->ubpr.unsol_bcast_probe_resp_interval &&
 	    nl80211_unsol_bcast_probe_resp(bss, msg, &params->ubpr) < 0)
 		goto fail;
 
-- 
2.45.2

